packages/monitor-v2/src/bot-oo-v3/index.ts

```diff
@@ -33,6 +33,7 @@ async function main() {
       await delay(params.pollingDelay);
     } else {
       await delay(5); // Set a delay to let the transports flush fully.
+      await waitForLogger(logger);
```

Comment:
We have to be very careful with this change because if any transport sets it to false and never sets it back, it could cause bots to hang, causing the hub to time out. Just looking at the existing [discord transport](https://github.com/UMAprotocol/protocol/blob/d314228f5bae1e7ba03eca7cc98fe20c6fb7d7d1/packages/financial-templates-lib/src/logger/DiscordTransport.ts#L106-L117) (which I think is the only bot to touch isFlushed) can end up in an infinite recursive error loop. We might want to make sure that it terminates after some finite number of calls or times out before adding this to all the bots.

Thoughts?

Reply:
We already use waitForLogger in the monitors package that covers all other versions of OO, except for OOv3 and monitors package still constitutes for most of alerts.

But I agree its better to have some control not allowing this to hang. One solution would be to have logger timeout (e.g. defaults to 5 minutes) that allows to return early from waitForLogger even if not all transports are flushed. Better to do this in #4607 as its already changing its behavior. And after #4609 we could also send alert to the dedicated log error channel that some transports had failed to flush.

Reply:
Added logger flush timeout in #4607

Reply:
LGTM once #4607 is merged!